File: 4045-longest-balanced-subarray-i/longest-balanced-subarray-i.cpp
class Solution {
public:
    int longestBalanced(vector<int>& nums) {
        int n = nums.size();

        int ans = 0;

        for (int i = 0; i <n ; i ++){
            unordered_set<int> even_set;
            unordered_set<int> odd_set;

            for (int j = i; j <n; j++){
                if (nums[j] % 2  == 0) even_set.insert(nums[j]);
                else odd_set.insert(nums[j]);

                if (even_set.size() == odd_set.size()) ans = max(ans, j-i + 1);
            }


        }
        return ans;

        
    }
};

File: 4045-longest-balanced-subarray-i/longest-balanced-subarray-i.py
class Solution(object):
    def longestBalanced(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        n = len(nums)
        out  = 0

        for i in range(0,n):
            even = set()
            odd = set()
            for j in range(i,n):
                if nums[j] %2  == 0:
                    even.add(nums[j])
                else:
                    odd.add(nums[j])
                
                if len(even) == len(odd):
                    out = max(out, j-i + 1)
        
        return out













        